viz/metrics-api/client/client.go

```diff
@@ -0,0 +1,200 @@
+	"github.com/linkerd/linkerd2/pkg/protohttp"
+	pb "github.com/linkerd/linkerd2/viz/metrics-api/gen/viz"
+	log "github.com/sirupsen/logrus"
+	"go.opencensus.io/plugin/ochttp"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	// APIRoot is the API's root path. It must be absolute (with a leading slash)
+	APIRoot = "/"
+
+	apiVersion = "v1"
+
+	// APIPrefix is the prefix all the API endpoints must share. It must be relative
+	// (without a leading slash)
+	APIPrefix = "api/" + apiVersion + "/"
+
+	apiPort       = 8085
+	apiDeployment = "linkerd-metrics-api"
+)
+
+type grpcOverHTTPClient struct {
+	serverURL  *url.URL
+	httpClient *http.Client
+	namespace  string
+}
+
+func (c *grpcOverHTTPClient) StatSummary(ctx context.Context, req *pb.StatSummaryRequest, _ ...grpc.CallOption) (*pb.StatSummaryResponse, error) {
+	var msg pb.StatSummaryResponse
+	err := c.apiRequest(ctx, "StatSummary", req, &msg)
+	return &msg, err
+}
+
+func (c *grpcOverHTTPClient) Edges(ctx context.Context, req *pb.EdgesRequest, _ ...grpc.CallOption) (*pb.EdgesResponse, error) {
+	var msg pb.EdgesResponse
+	err := c.apiRequest(ctx, "Edges", req, &msg)
+	return &msg, err
+}
+
+func (c *grpcOverHTTPClient) TopRoutes(ctx context.Context, req *pb.TopRoutesRequest, _ ...grpc.CallOption) (*pb.TopRoutesResponse, error) {
+	var msg pb.TopRoutesResponse
+	err := c.apiRequest(ctx, "TopRoutes", req, &msg)
+	return &msg, err
+}
+
+func (c *grpcOverHTTPClient) Gateways(ctx context.Context, req *pb.GatewaysRequest, _ ...grpc.CallOption) (*pb.GatewaysResponse, error) {
+	var msg pb.GatewaysResponse
+	err := c.apiRequest(ctx, "Gateways", req, &msg)
+	return &msg, err
+}
+
+func (c *grpcOverHTTPClient) SelfCheck(ctx context.Context, req *healthcheckPb.SelfCheckRequest, _ ...grpc.CallOption) (*healthcheckPb.SelfCheckResponse, error) {
+	var msg healthcheckPb.SelfCheckResponse
+	err := c.apiRequest(ctx, "SelfCheck", req, &msg)
+	return &msg, err
+}
+
+func (c *grpcOverHTTPClient) ListPods(ctx context.Context, req *pb.ListPodsRequest, _ ...grpc.CallOption) (*pb.ListPodsResponse, error) {
+	var msg pb.ListPodsResponse
+	err := c.apiRequest(ctx, "ListPods", req, &msg)
+	return &msg, err
+}
+
+func (c *grpcOverHTTPClient) ListServices(ctx context.Context, req *pb.ListServicesRequest, _ ...grpc.CallOption) (*pb.ListServicesResponse, error) {
+	var msg pb.ListServicesResponse
+	err := c.apiRequest(ctx, "ListServices", req, &msg)
+	return &msg, err
+}
+
+func (c *grpcOverHTTPClient) Tap(ctx context.Context, req *pb.TapRequest, _ ...grpc.CallOption) (pb.Api_TapClient, error) {
+	return nil, status.Error(codes.Unimplemented, "Tap is deprecated in public API, use tap APIServer")
+}
+
+func (c *grpcOverHTTPClient) TapByResource(ctx context.Context, req *pb.TapByResourceRequest, _ ...grpc.CallOption) (pb.Api_TapByResourceClient, error) {
+	return nil, status.Error(codes.Unimplemented, "Tap is deprecated in public API, use tap APIServer")
+}
```

Comment:
We should remove the "public" part from these. Maybe even we can get rid of them now?

Reply:
Indeed. I've just pushed the changes related to that removal :+1: